src/ui/view/dataverse/component/generalMetadata/generalMetadataValue.tsx

```diff
@@ -0,0 +1,51 @@
+import { useTranslation } from 'react-i18next'
+import * as E from 'fp-ts/Either'
+import { pipe } from 'fp-ts/lib/function'
+import { fromPredicate } from 'fp-ts/lib/Either'
+import type { ItemGeneralMetadata } from '@/ui/view/dataverse/types'
+import './generalMetadata.scss'
+import './i18n/index'
+import { DateFromTo } from './dateFromTo'
+
+const isPeriodGeneralMetadata = (
+  metadata: ItemGeneralMetadata
+): metadata is ItemGeneralMetadata & { property: 'period'; value: [string, string] } =>
+  metadata.property === 'period'
+
+const isGeneralMetadata = (
+  metadata: ItemGeneralMetadata
+): metadata is ItemGeneralMetadata & { value: string } => metadata.property !== 'period'
+
+type MetadataValueProps = {
+  metadata: ItemGeneralMetadata
+}
+
+export const GeneralMetadataValue = ({ metadata }: MetadataValueProps): JSX.Element | null => {
+  const namespace = 'generalMetadata'
+  const { t, i18n } = useTranslation(namespace)
+  const { value, property } = metadata
+  if (isGeneralMetadata(metadata)) {
+    const translationKeyExists = i18n.exists(`${namespace}.topic.value.${value}`, {
+      ns: namespace,
+      lng: i18n.language
+    })
+    const translatedValue = pipe(
+      value,
+      fromPredicate(
+        () => translationKeyExists,
+        () => value
+      ),
+      E.map(val => t(`${namespace}.${property}.value.${val}`)),
+      E.getOrElse(() => value)
+    )
+    return <span>{translatedValue}</span>
+  }
```

Comment:
A bit puzzling to me, particularly due to the utilization of "E.fromPredicate" which is surprising. Why not passing in a default value for cases the key could not be found?


```suggestion
  if (isGeneralMetadata(metadata)) {
    return <span>{t(`${namespace}.${property}.value.${value}`, `${value}`)}</span>
  }
```

Reply:
Thanks for pointing this out! Didn't know about the second i18next parameter that acts as a fallback. Usage of `fromPredicate` was not relevant nor necessary in that case.